src/dra/6D59C.c

```diff
@@ -797,7 +797,7 @@ s32 CheckChainLimit(s32 itemId, s32 handId) {
         // Hack to load unkAE as an s16 (struct has s8)
         // Longer term, figure out what g_Entites[16-64] are
         // and make dedicated ent extension.
-        if (LOH(entity->ext.generic.unkAE) != itemId) {
+        if (entity->ext.weapon.equipId != itemId) {
```

Comment:
Does weapon.equipId align for all the subweapons? Seems like we want static asserts

Reply:
This isn't subweapons, this is real weapons. Buffalo stars, etc

Reply:
The issue still exists there. We have the anim asserts which mostly covers the problem probably https://github.com/Xeeynamo/sotn-decomp/blob/887e32510bdcaa81acc11748898a4db0d691b405/include/entity.h#L1971